components/backends/vllm/src/dynamo/vllm/engine_monitor.py

```diff
@@ -43,13 +43,26 @@ def __init__(self, runtime: DistributedRuntime, engine_client: AsyncLLM):
     def __del__(self):
         self._monitor_task.cancel()
 
+    def _shutdown_engine(self):
+        """
+        Shutdown the vLLM engine on crash scenarios, in order to free resources.
+        """
+        try:
+            logger.info("Shutting down vLLM engine to free GPU memory...")
+            self.engine_client.shutdown()
+            logger.info("vLLM engine shutdown completed")
+        except Exception as e:
+            logger.warning(f"Engine shutdown method failed: {e}")
+
     async def _check_engine_health(self):
         while True:
             try:
                 await self.engine_client.check_health()
                 await asyncio.sleep(HEALTH_CHECK_INTERVAL)
             except EngineDeadError as e:
                 logger.error(f"vLLM AsyncLLM health check failed: {e}")
+                logger.warning("Initiating engine cleanup.")
```

Comment:
This is a good example of too many logs statements! :-)  Remove the new warning.